// node-capmonster/dist/tasks/fun_captcha.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FuncaptchaTask = void 0;
const capmonster_1 = require("../capmonster");
class FuncaptchaTask extends capmonster_1.UAProxy {
    constructor(clientKey) {
        super(clientKey);
    }
    createTask = async (websiteUrl, websitePublicKey, apiJsSubdomain, dataBlob, cookies, noCache) => {
        console.warn("`createTask` is deprecated, use `task` & `createWithTask` to avoid errors in future versions");
        const data = {
            type: "FunCaptchaTask",
            websiteURL: websiteUrl,
            websitePublicKey,
            funcaptchaApiJSSubdomain: apiJsSubdomain,
            cookies: cookies ? this.addCookies(cookies) : undefined,
            data: dataBlob,
            noCache,
        };
        const [proxyData] = this.isProxyTask(data);
        const [userAgentData] = this.isUserAgentTask(proxyData);
        return await this._createTask(userAgentData);
    };
    task = (task) => task;
    createWithTask = async (task) => {
        const data = {
            type: "FunCaptchaTask",
            ...task,
        };
        const [proxyData] = this.isProxyTask(data);
        const [userAgentData] = this.isUserAgentTask(proxyData);
        return await this._createTask(userAgentData);
    };
    getTaskResult = async (taskId) => this._getTaskResult(taskId);
    joinTaskResult = async (taskId, timeout) => this._joinTaskResult(taskId, timeout);
}
exports.FuncaptchaTask = FuncaptchaTask;
