// node-capmonster/dist/tasks/geetest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeeTestTask = void 0;
const capmonster_1 = require("../capmonster");
class GeeTestTask extends capmonster_1.UAProxy {
    constructor(clientKey) {
        super(clientKey);
    }
    createTask = async (websiteUrl, gt, challenge, apiServerSubdomain, getLib) => {
        console.warn("This function is deprecated, use `task` & `createWithTask` to avoid errors in future versions");
        const data = {
            type: "GeeTestTask",
            websiteURL: websiteUrl,
            gt,
            challenge,
            geetestApiServerSubdomain: apiServerSubdomain,
            geetestGetLib: getLib,
        };
        const [proxyData] = this.isProxyTask(data);
        const [userAgentData] = this.isUserAgentTask(proxyData);
        return await this._createTask(userAgentData);
    };
    task = (task) => task;
    createWithTask = async (task) => {
        const data = {
            type: "GeeTestTask",
            ...task,
        };
        const [proxyData] = this.isProxyTask(data);
        const [userAgentData] = this.isUserAgentTask(proxyData);
        return await this._createTask(userAgentData);
    };
    getTaskResult = async (taskId) => this._getTaskResult(taskId);
    joinTaskResult = async (taskId, timeout) => this._joinTaskResult(taskId, timeout);
}
exports.GeeTestTask = GeeTestTask;
