// node-capmonster/dist/tasks/hcaptcha.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HCaptchaTask = void 0;
const capmonster_1 = require("../capmonster");
class HCaptchaTask extends capmonster_1.UAProxy {
    constructor(clientKey) {
        super(clientKey);
    }
    createTask = async (websiteUrl, websiteKey, isInvisible, customData, cookies, noCache) => {
        console.warn("This function is deprecated, use `task` & `createWithTask` to avoid errors in future versions");
        const data = {
            type: "HCaptchaTask",
            websiteURL: websiteUrl,
            websiteKey,
            isInvisible,
            data: customData,
            cookies: cookies ? this.addCookies(cookies) : undefined,
            noCache,
        };
        const [proxyData] = this.isProxyTask(data);
        return await this._createTask(proxyData);
    };
    task = (task) => task;
    createWithTask = async (task) => {
        const data = {
            type: "HCaptchaTask",
            ...task,
        };
        const [proxyData] = this.isProxyTask(data);
        return await this._createTask(proxyData);
    };
    getTaskResult = async (taskId) => this._getTaskResult(taskId);
    joinTaskResult = async (taskId, timeout) => this._joinTaskResult(taskId, timeout);
}
exports.HCaptchaTask = HCaptchaTask;
