// node-capmonster/dist/tasks/recaptcha_v2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecaptchaV2Task = void 0;
const capmonster_1 = require("../capmonster");
class RecaptchaV2Task extends capmonster_1.UAProxy {
    constructor(clientKey) {
        super(clientKey);
    }
    createTask = async (websiteUrl, websiteKey, cookies, recaptchaSValue, noCache) => {
        console.warn("This function is deprecated, use `task` & `createWithTask` to avoid errors in future versions");
        const data = {
            type: "NoCaptchaTask",
            websiteURL: websiteUrl,
            websiteKey,
            cookies: cookies ? this.addCookies(cookies) : undefined,
            recaptchaDataSValue: recaptchaSValue,
            noCache,
        };
        const [proxyData] = this.isProxyTask(data);
        const [userAgentData] = this.isUserAgentTask(proxyData);
        return await this._createTask(userAgentData);
    };
    task = (task) => task;
    createWithTask = async (task) => {
        const data = {
            type: "NoCaptchaTask",
            ...task,
        };
        const [proxyData] = this.isProxyTask(data);
        const [userAgentData] = this.isUserAgentTask(proxyData);
        return await this._createTask(userAgentData);
    };
    getTaskResult = async (taskId) => this._getTaskResult(taskId);
    joinTaskResult = async (taskId, timeout) => this._joinTaskResult(taskId, timeout);
}
exports.RecaptchaV2Task = RecaptchaV2Task;
